sw/device/lib/dif/dif_uart.h

```diff
@@ -0,0 +1,283 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef _SW_DEVICE_LIB_DIF_UART_H_
+#define _SW_DEVICE_LIB_DIF_UART_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+/**
+ * UART interrupt configuration
+ *
+ * Enumeration used to enable, disable, test and query the UART interrupts.
+ * Please see the comportability specification for more information:
+ * https://docs.opentitan.org/doc/rm/comportability_specification/
+ */
+typedef enum dif_uart_interrupt {
+  kDifUartInterruptTxWatermark = 0, /**< TX FIFO dipped below watermark */
+  kDifUartInterruptRxWatermark,     /**< RX FIFO reached high watermark */
+  kDifUartInterruptTxEmpty,         /**< TX FIFO empty */
+  kDifUartInterruptRxOverflow,      /**< RX FIFO overflow */
+  kDifUartInterruptRxFrameErr,      /**< RX framing error */
+  kDifUartInterruptRxBreakErr,      /**< RX break condition */
+  kDifUartInterruptRxTimeout, /**< RX FIFO timeout (not empty in a set time) */
+  kDifUartInterruptRxParityErr, /**< RX parity error detection */
+} dif_uart_interrupt_t;
+
+/**
+ * TX and RX FIFO depth configuration
+ *
+ * Enumeration used to set the upper limit of bytes to queue.
+ */
+typedef enum dif_uart_watermark {
+  kDifUartWatermarkByte1 = 0, /**< 1 byte */
+  kDifUartWatermarkByte4,     /**< 4 bytes */
+  kDifUartWatermarkByte8,     /**< 8 bytes */
+  kDifUartWatermarkByte16,    /**< 16 bytes */
+  kDifUartWatermarkByte30,    /**< 30 bytes */
+} dif_uart_watermark_t;
+
+/**
+ * Generic enable/disable enumeration
+ *
+ * Enumeration used to enable/disable bits, flags, ...
+ */
+typedef enum dif_uart_enable {
+  kDifUartEnable = 0, /**< enable */
+  kDifUartDisable,    /**< disable */
+} dif_uart_enable_t;
+
+/**
+ * Generic odd/even enumeration
```

Comment:
I think this is only used for setting parity, may be we should not call it "Generic".

Reply:
I think parity is generic enough. It only is used in this code to chose the UART parity control, but can be used for other odd/even operations. The expectation is that this eventually live in it's own header file under ../base.

Reply:
Drop the generic line here, given it's `kDifUartParityX` now.

Reply:
done